subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -943,100 +872,104 @@ void nct_uninit(void)
 	mqtt_client_initialized = false;
 }
 
-#if defined(CONFIG_NRF_CLOUD_STATIC_IPV4)
-int nct_connect(void)
+
+static void nct_init_mqtt(void)
+{
+	if (!mqtt_client_initialized) {
+		mqtt_client_init(&nct.client);
+
+		nct.client.evt_cb = nct_mqtt_evt_handler;
+		nct.client.client_id.utf8 = (uint8_t *)client_id_ptr;
+		nct.client.client_id.size = strlen(client_id_ptr);
+		nct.client.protocol_version = MQTT_VERSION_3_1_1;
+		nct.client.password = NULL;
+		nct.client.user_name = NULL;
+		nct.client.keepalive = CONFIG_NRF_CLOUD_MQTT_KEEPALIVE;
+		nct.client.clean_session = persistent_session ? 0U : 1U;
+		LOG_DBG("MQTT clean session flag: %u",
+			nct.client.clean_session);
+
+#if defined(CONFIG_MQTT_LIB_TLS)
+		nct.client.transport.type = MQTT_TRANSPORT_SECURE;
+		nct.client.rx_buf = nct.rx_buf;
+		nct.client.rx_buf_size = sizeof(nct.rx_buf);
+		nct.client.tx_buf = nct.tx_buf;
+		nct.client.tx_buf_size = sizeof(nct.tx_buf);
+
+		struct mqtt_sec_config *tls_config =
+			   &nct.client.transport.tls.config;
+		memcpy(tls_config, &nct.tls_config,
+			   sizeof(struct mqtt_sec_config));
+#else
+		nct.client.transport.type = MQTT_TRANSPORT_NON_SECURE;
+#endif
+		mqtt_client_initialized = true;
+	}
+}
+
+static int nct_try_address(struct sockaddr *const addr)
```

Comment:
```suggestion
static int nct_mqtt_connect(struct sockaddr *const addr)
```

Reply:
Better stick with `nct_connect_host_cb` in keeping with your other suggestions